main.go

```diff
@@ -576,6 +582,38 @@ func createDatabase(project, instance, dbName string, conv *internal.Conv, out *
 	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, dbName), nil
 }
 
+func UpdateDDLForeignKeys(project, instance, dbName string, conv *internal.Conv, out *os.File) error {
+	fmt.Fprintf(out, "Updating database %s in instance %s with default permissions ...\n", dbName, instance)
+	ctx := context.Background()
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("can't create admin client: %w\n", analyzeError(err, project, instance))
+	}
+	defer adminClient.Close()
+	// The schema we send to Spanner excludes comments (since Cloud
+	// Spanner DDL doesn't accept them), and protects table and col names
+	// using backticks (to avoid any issues with Spanner reserved words).
+	fkStmts := conv.GetFK(ddl.Config{Comments: false, ProtectIds: true, ForeignKeys: true})
+	for _, fkStmt := range fkStmts {
+		op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+			Database:   fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, dbName),
+			Statements: []string{fkStmt},
+		})
+		if err != nil {
+			fmt.Printf("Can't add foreign key with statement %s: %s\n", fkStmt, err)
+			conv.Unexpected(fmt.Sprintf("Can't add foreign key with statement %s: %s", fkStmt, err))
+			continue
+		}
+		if err := op.Wait(ctx); err != nil {
```

Comment:
Agreed with Asheesh: we probably need to do more here to communicate what is happening.

I think there are two separate issues:
a) the UpdateDatabaseDdl operations themselves will take some time.
b) these operations will trigger long-running schema operations e.g. for backfill of referencing and referenced indexes.

I just did some testing for (a): it looks like each UpdateDatabaseDdl operation can take from 15 seconds to 100 seconds each (on average about 25 seconds).
This was for empty tables, where each UpdateDatabaseDdl operation was very simple (a single constraint with one col), and essentially identical (just for a different tables).

We already have a progress.go module for reporting progress which we use elsewhere e.g. for reporting progress on loading data. Let's just use that: its internal/progress.go.  Here's how we could use it:

```
func UpdateDDLForeignKeys(project, instance, dbName string, conv *internal.Conv, out *os.File) error {
	ctx := context.Background()
	adminClient, err := database.NewDatabaseAdminClient(ctx)
	if err != nil {
		return fmt.Errorf("can't create admin client: %w\n", analyzeError(err, project, instance))
	}
	defer adminClient.Close()
	// The schema we send to Spanner excludes comments (since Cloud
	// Spanner DDL doesn't accept them), and protects table and col names
	// using backticks (to avoid any issues with Spanner reserved words).
	fkStmts := conv.GetFK(ddl.Config{Comments: false, ProtectIds: true, ForeignKeys: true})
	msg := fmt.Sprintf("Updating schema of database %s in instance %s with foreign key constraints ...", dbName, instance)
	p := internal.NewProgress(int64(len(fkStmts)), msg, internal.Verbose())
	for i, fkStmt := range fkStmts {
		op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
			Database:   fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, dbName),
			Statements: []string{fkStmt},
		})
		if err != nil {
			fmt.Printf("Can't add foreign key with statement %s: %s\n", fkStmt, err)
			conv.Unexpected(fmt.Sprintf("Can't add foreign key with statement %s: %s", fkStmt, err))
			continue
		}
		if err := op.Wait(ctx); err != nil {
			fmt.Printf("Can't add foreign key with statement %s: %s\n", fkStmt, err)
			conv.Unexpected(fmt.Sprintf("Can't add foreign key with statement %s: %s", fkStmt, err))
		}
		internal.VerbosePrintln("Updated schema with statement: " + fkStmt)
		p.MaybeReport(int64(i + 1))
	}
	p.Done()
	return nil
}
```



// TODO:

Reply:
Sure nevin, makes sense to show user what is happening, updated code as mentioned.

Reply:
Hi Pradip, can you please add the TODO as Nevin mentioned in his comment. This is definitely something we would look to fix in future.

For reference, the TODO mentioned above was:

// TODO: Improve performance of the foreign key constraint updates.
// For example, issue all of the update ops first before waiting for them to complete
// so that that can execute in parallel. We could also print out ids of the
// long-running operations.

Reply:
I see that the TODO was added, resolving the comment.